server/app.py

```diff
@@ -104,38 +181,105 @@ async def submit(request: SubmitRequest):
     Returns:
         Response[JobCreated]: The response model for a job created.
     """
+
     # TODO: get task by name, e.g. request.task_name
     task = process_file
     serialized_agent = pickle.dumps(request.agent)
-    logger.debug(f"Submitting task {task.name} with agent {serialized_agent}")
+
+    logger.info(f"Submitting task {task.name} with agent {serialized_agent}")
     result = task.delay(serialized_agent=serialized_agent)
-    logger.debug(f"Task {task.name} submitted with job_id {result.id}")
+    logger.info(f"Task {task.name} submitted with job_id {result.id}")
+
     return Response[JobCreated](data=JobCreated(job_id=result.id))
 
 
-class Status(Enum):
-    PENDING = "Pending"
-    INPROGRESS = "InProgress"
-    COMPLETED = "Completed"
-    FAILED = "Failed"
-    CANCELED = "Canceled"
+async def poll_for_streaming_results(job_id: str, batch_size: int, handler: Callable[List[dict], None]):
+    """
+    Poll for results in the kafka output topic and handle them.
+    """
+    logger.info(f"Polling for results {job_id=}")
+
+    consumer = AIOKafkaConsumer(
+        f"adala-output-{job_id}",
+        bootstrap_servers=settings.kafka_bootstrap_servers,
+        value_deserializer=lambda v: json.loads(v.decode("utf-8")),
+        auto_offset_reset="earliest",
+        # group_id="adala-consumer-group",
+    )
+    await consumer.start()
+    logger.info(f"consumer started {job_id=}")
 
+    try:
+        data = await consumer.getmany(timeout_ms=30_000, max_records=batch_size)
+        logger.info(f"got batch {data=}")
+        for tp, messages in data.items():
+            assert tp.topic == f"adala-output-{job_id}", f"Unexpected topic: {tp.topic}"
```

Comment:
I don't think we need this check, since we're the ones setting the topic name above. Is there a situation where this would be a different topic name?